org.kicad.KiCad.yml

```diff
@@ -143,45 +139,58 @@ modules:
 - name: kicad
   buildsystem: cmake-ninja
   config-opts:
-  - -DCMAKE_BUILD_TYPE=Release
-  - -DGLEW_INCLUDE_DIR=/app/include/GL
+  - -DCMAKE_BUILD_TYPE=RelWithDebInfo
   - -DOCC_INCLUDE_DIR=/app/include/opencascade
   - -DOPENGL_glu_LIBRARY=/app/lib/libGLU.so
   - -DKICAD_BUILD_QA_TESTS=OFF
-  - -DKICAD_SCRIPTING_PYTHON3=ON
+  - -DKICAD_BUILD_I18N=ON
+  - -DKICAD_USE_EGL=ON
   - -DKICAD_USE_OCC=ON
   - -DKICAD_USE_OCE=OFF
-  - -DKICAD_SCRIPTING_WXPYTHON_PHOENIX=ON
+  - -DKICAD_LIBRARY_DATA=/app/extensions/Library
   post-install:
-  - install -d /app/extensions/Library
+  - for f in $FLATPAK_DEST/share/applications/*.desktop; do
+      desktop_name="$(basename $f)";
+      desktop_app_name="${desktop_name%.desktop}";
+      app_name="${desktop_app_name#org.kicad.}";
+      if [ "$app_name" = "kicad" ]; then
+        desktop-file-edit --set-icon="${FLATPAK_ID}" --set-key=X-Flatpak-RenamedFrom --set-value="${desktop_name};" "$f";
+        mv $FLATPAK_DEST/share/applications/{org.kicad.kicad,$FLATPAK_ID}.desktop;
+      else
+        desktop-file-edit --set-icon="${FLATPAK_ID}.${app_name}" --set-key=X-Flatpak-RenamedFrom --set-value="${desktop_name};" "$f";
+        mv $FLATPAK_DEST/share/applications/{org.kicad,$FLATPAK_ID}.${app_name}.desktop;
+      fi;
+    done;
+    unset desktop_name desktop_app_name app_name
+  - sed -i "s/\(application-x-kicad-.\+\)/$FLATPAK_ID.\1/" $FLATPAK_DEST/share/mime/packages/kicad-kicad.xml
```

Comment:
Why are you renaming the mime info for KiCad? That is supposed to be the MIME info for the file, so all KiCad packages should use the same MIME type for the files.

Reply:
Flathub requires exported MIME types to start with the flatpak app ID, or they will be deleted from the flatpak (resulting in no MIME association).

Reply:
This was an issue for a long time. In the 5.1.x time I had a patch for this (deleted with this PR), but now I switched to this build script to be more flexible.

Reply:
See #13

Reply:
That truly seems like a bad design decision by them then. The MIME standard explicitly says that MIME types can only start with one of 5 possible identifiers (https://www.rfc-editor.org/rfc/rfc2046.html#section-3), so starting with the Flatpak ID is not standards compliant.

Reply:
I'm not sure if this was clear before given my comments were sent from mobile, but this is not renaming the MIME types, but instead *prepending* the *icon associations (icon file names)*. This is what flatpaks require. Before #13, installing KiCad via flathub would leave the user with broken file associations, i.e. double-clicking on any KiCad file would open it in a text editor instead of in the correct application, and the icons were not showing up. So, IANAL, but I think this is still standards compliant.

Researching the reasoning behind this requirement is a little difficult, as this part is unfortunately underspecified in the flatpak documentation [1], but the bottom line is that this facilitates sandboxing in that it doesn't let a rogue app install MIME type associations for other apps’ file types. An app can only install associations to their own custom file types this way. `flatpak` does mangle MIME types even more in the final build/install stages for the same reasons (for example removing magic byte info and setting priorities to low percentage values), but this happens automatically and cannot be circumvented at all.

In the end, we want the file type associations to work for users, and this is the way how to it with flatpaks.

[1]: I found this ticket to be the best description in lack of proper documentation on their end: https://github.com/flatpak/flatpak/issues/1572

FWIW, this is how the MIME type definitions end up in the final flatpak (gerbers look similar). As you can see, the MIME type tags are still intact:

```xml
$ cat /app/share/mime/packages/org.kicad.KiCad-kicad.xml 
<?xml version="1.0" encoding="UTF-8"?>
<mime-info xmlns="http://www.freedesktop.org/standards/shared-mime-info">
  <mime-type type="application/x-kicad-project">
    <sub-class-of type="text/plain"/>
    <comment>KiCad Project</comment>
    <comment xml:lang="zh_TW">KiCad 專案</comment>
    <comment xml:lang="zh_CN">KiCad 项目</comment>
    <comment xml:lang="tr">KiCad Projesi</comment>
    <comment xml:lang="th">โครงการ KiCad</comment>
    <comment xml:lang="sr">KiCad пројект</comment>
    <comment xml:lang="ru">Проект KiCad</comment>
    <comment xml:lang="pt_br">Projeto KiCad</comment>
    <comment xml:lang="pt">Projeto KiCad</comment>
    <comment xml:lang="pl">Projekt KiCad</comment>
    <comment xml:lang="nl">KiCad-project</comment>
    <comment xml:lang="ko">KiCad 프로젝트</comment>
    <comment xml:lang="ja">Kicad プロジェクト ファイル</comment>
    <comment xml:lang="id">Proyek KiCad</comment>
    <comment xml:lang="fr">Projet KiCad</comment>
    <comment xml:lang="fi">KiCad-projekti</comment>
    <comment xml:lang="es_MX">Proyecto de KiCad</comment>
    <comment xml:lang="es">Proyecto de KiCad</comment>
    <comment xml:lang="en">KiCad Project</comment>
    <comment xml:lang="el">Έργο KiCad</comment>
    <comment xml:lang="de">KiCad-Projekt</comment>
    <comment xml:lang="cs">KiCad projekt</comment>
    <generic-icon name="org.kicad.KiCad.application-x-kicad-project"/>
    <glob pattern="*.pro" weight="40"/>
    <glob pattern="*.kicad_pro" weight="50"/>
  </mime-type>
  <mime-type type="application/x-kicad-schematic">
    <sub-class-of type="text/plain"/>
    <comment>KiCad Schematic</comment>
    <comment xml:lang="zh_TW">KiCad 原理圖</comment>
    <comment xml:lang="zh_CN">KiCad 原理图</comment>
    <comment xml:lang="tr">KiCad Şeması</comment>
    <comment xml:lang="th">แผนผังวงจร KiCad</comment>
    <comment xml:lang="ru">Схема KiCad</comment>
    <comment xml:lang="ro">Schematică KiCad</comment>
    <comment xml:lang="pt_br">Esquema KiCad</comment>
    <comment xml:lang="pt">Esquema KiCad</comment>
    <comment xml:lang="pl">Schemat KiCad</comment>
    <comment xml:lang="nl">KiCad schema</comment>
    <comment xml:lang="ko">KiCad 회로도</comment>
    <comment xml:lang="ja">KiCad 回路図ファイル</comment>
    <comment xml:lang="id">Skematik KiCad</comment>
    <comment xml:lang="fr">Schématique KiCad</comment>
    <comment xml:lang="fi">KiCad-kytkentäkaavio</comment>
    <comment xml:lang="es_MX">Esquema de KiCad</comment>
    <comment xml:lang="es">Esquema de KiCad</comment>
    <comment xml:lang="en">KiCad Schematic</comment>
    <comment xml:lang="el">Σχηματικό KiCad</comment>
    <comment xml:lang="de">KiCad-Schaltplan</comment>
    <comment xml:lang="cs">KiCad schéma</comment>
    <generic-icon name="org.kicad.KiCad.application-x-kicad-schematic"/>
    <glob pattern="*.sch" weight="40"/>
    <glob pattern="*.kicad_sch" weight="50"/>
    <magic priority="50">
      <!--The new schematic file (.kicad_sch) header-->
      <match type="string" offset="0" value="(kicad_sch"/>
    </magic>
    <magic priority="40">
      <!--The old schematic file (.sch) header-->
      <match type="string" offset="0" value="EESchema Schematic File"/>
    </magic>
  </mime-type>
  <mime-type type="application/x-kicad-pcb">
    <sub-class-of type="text/plain"/>
    <comment>KiCad Printed Circuit Board</comment>
    <comment xml:lang="zh_TW">KiCad 印刷電路板</comment>
    <comment xml:lang="zh_CN">KiCad 印刷电路板</comment>
    <comment xml:lang="tr">KiCad Baskılı Devre Kartı</comment>
    <comment xml:lang="th">แผ่นวงจรพิมพ์ KiCad</comment>
    <comment xml:lang="sv">KiCad-mönsterkort</comment>
    <comment xml:lang="ru">Печатная плата KiCad</comment>
    <comment xml:lang="ro">Placă de circuite imprimate KiCad</comment>
    <comment xml:lang="pt_br">Placa de circuito impresso KiCad</comment>
    <comment xml:lang="pt">Placa de circuito impresso KiCad</comment>
    <comment xml:lang="pl">Obwód drukowany KiCad</comment>
    <comment xml:lang="nl">KiCad PCB (Printed Circuit Board)</comment>
    <comment xml:lang="ko">KiCad 인쇄 회로 기판</comment>
    <comment xml:lang="ja">KiCad プリント基板ファイル</comment>
    <comment xml:lang="id">Papan Sirkuit KiCad</comment>
    <comment xml:lang="fr">Fichier Circuit Imprimé KiCad</comment>
    <comment xml:lang="fi">KiCad-piirilevy</comment>
    <comment xml:lang="es_MX">Placa de circuito impreso de KiCad</comment>
    <comment xml:lang="es">Placa de circuito impreso de KiCad</comment>
    <comment xml:lang="en">KiCad Printed Circuit Board</comment>
    <comment xml:lang="el">Πλακέτα Τυπωμένου Κυκλώματος KiCad</comment>
    <comment xml:lang="de">KiCad-Leiterplatte</comment>
    <comment xml:lang="cs">KiCad deska plošných spojů</comment>
    <generic-icon name="org.kicad.KiCad.application-x-kicad-pcb"/>
    <glob pattern="*.kicad_pcb" weight="50"/>
    <magic priority="50">
      <!--The new board file (.kicad_pcb) header-->
      <match type="string" offset="0" value="(kicad_pcb"/>
    </magic>
    <magic priority="40">
      <!--The old board file (.brd) header-->
      <match type="string" offset="0" value="PCBNEW-BOARD"/>
    </magic>
  </mime-type>
</mime-info>
```